File: core-scala/src/main/scala/com/roboxue/niffler/NifflerSession.scala
package com.roboxue.niffler

import scala.collection.mutable

class NifflerSession {
  private val storage: mutable.Map[Token[_], Any] = mutable.Map.empty

  def get[T](token: Token[T]): T = {
    storage(token).asInstanceOf[T]
  }

  def set[T](token: Token[T], value: T): Unit = {
    storage(token) = value
  }
}

File: core-python/niffler/NifflerSession.py
class NifflerSession:
    __storage__ = None  # type: Dict[Token, Any]

    def __init__(self):
        self.__storage__ = {}

    def get(self, token):
        return self.__storage__[token]

    def set(self, token, value):
        self.__storage__[token] = value